Автоматизированные тесты для финального проекта Яндекс.Практикум.

Pinky-Pinky opened the pull request:
Добавлены тесты, конфигурация и pom.xml

[Тестируемый сайт Яндекс.Самокат](https://qa-scooter.praktikum-services.ru/)

Commit:
Добавлены тесты, конфигурация и pom.xml

Commit:
Добавлены тесты для страницы заказа

okryukova commented on .idea/jarRepositories.xml:
⛔️Нужно исправить.  Папку .idea не нужно было загружать в репозиторий. Эта папка должна быть добавлена в .gitignore.

okryukova commented on src/main/java/pageObjects/MainPage.java:
```diff
@@ -0,0 +1,98 @@
+package pageObjects;
```
⛔️Нужно исправить. Необходимо исправить наименование пакетов в соответствии с правилами нейминга. Пакеты — допустимы только строчные буквы. В случае, если слов несколько, их следует разделять точкой. Префикс должен быть одним из доменных имен верхнего уровня, таких как com, ru и т.д. Наименование пакета должно отражать суть содержимого. https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

okryukova commented on src/main/java/pageObjects/OrderPage.java:
```diff
@@ -0,0 +1,183 @@
+    public void selectMetroStation(String station) {
+        metroInput.sendKeys(station);
+        WebElement stationElement = wait.until(ExpectedConditions.elementToBeClickable(
+                By.xpath("//div[normalize-space(text())='" + station + "']")
```
⛔️Нужно исправить. Статическую часть локатора лучше хранить в поле класса
⚠️Можно улучшить.  Для формирования локатора можно использовать string.format

okryukova commented on src/main/java/pageObjects/OrderPage.java:
```diff
@@ -0,0 +1,183 @@
+        ((JavascriptExecutor) driver).executeScript("arguments[0].click();", button);
+
+        // Ожидаем появление модального окна
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//div[contains(@class, 'Order_Modal__YZ-d3')]")));
```
⛔️Нужно исправить. Локаторы нужно хранить в полях класса. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/ui/MainPageAccordionTests.java:
```diff
@@ -0,0 +1,79 @@
+    @Parameterized.Parameters(name = "Текст в блоке \"Вопросы о важном\". Проверяемый элемент: {1}")
+    public static Object[][] setTestData() {
+        return new Object[][]{
+                {0, "Сколько это стоит? И как оплатить?", "Сутки — 400 рублей. Оплата курьеру — наличными или картой."},
```
⚠️Можно улучшить. Тестовые данные  должны содержаться либо в константах тестового класса, либо в отдельном файле

okryukova commented on src/test/java/ui/MainPageAccordionTests.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    @Before
+    public void startUp() {
```
⚠️Можно улучшить. Блоки закрытия и открытия  браузера одинаковы для всех тестовых классов. Целесообразно вынести их в базовый или утилитный класс.

okryukova commented on src/test/java/ui/MainPageAccordionTests.java:
```diff
@@ -0,0 +1,79 @@
+        String actualItemText = mainPage.getAccordionItemText(numberOfElement);
+
+        // Проверки через Hamcrest
+        MatcherAssert.assertThat("Неверный текст в заголовке", actualHeaderText, equalTo(expectedHeaderText));
```
⚠️Можно улучшить.  Тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/ui/OrderPageTests.java:
```diff
@@ -0,0 +1,148 @@
+    }
+
+    @Test
+    public void orderWithHeaderButtonWhenSuccess() {
```
⚠️Можно улучшить. В названии метода необходимо использовать слово Test. Проверь, соблюдается ли данное правило для всех созданных тестов.

okryukova commented on .gitignore:
⚠️Можно улучшить. Хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройку (если необходима) и как запускать (mvn clean test - кавычки позволяют выполнять запуск из файла README.md). Для оформления данных в виде таблицы используй сочетание клавиш Ctrl + N или Cmd + N. Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

Commit:
Удалил папку .idea из Git

Commit:
Игнорируем .idea и target

Commit:
Новые изменения

Commit:
Дополнительные изменения

# Conflicts:
#	src/main/java/ru/yandex/praktikum/ui/pages/OrderPage.java
#	target/classes/pageObjects/OrderPage.class
#	target/surefire-reports/TEST-ui.MainPageAccordionTests.xml
#	target/surefire-reports/TEST-ui.MainPageLinksTests.xml
#	target/surefire-reports/TEST-ui.OrderPageTests.xml
#	target/surefire-reports/ui.MainPageAccordionTests.txt
#	target/surefire-reports/ui.MainPageLinksTests.txt
#	target/surefire-reports/ui.OrderPageTests.txt
#	target/test-classes/ui/OrderPageTests.class

okryukova commented on src/main/java/ru/yandex/praktikum/ui/pages/OrderPage.java.save:
⛔️Нужно исправить.  Неиспользуемый код нужно удалять

okryukova commented on target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst:
⛔️Нужно исправить.  Папку terget добавлять в пулл реквест не нужно
